### PR TITLE
Test wait for update

### PR DESCRIPTION
**Pull request by mogren**
*Issue #, if available:*

*Description of changes:*
* Run integration tests before and after updating the CNI
* Run e2e tests from CircleCI (part of #686)
* Merge config files from release-1.6 branch

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/lib/cluster.sh [hunk 1]
```diff
@@ -1,15 +1,19 @@
 #!/usr/bin/env bash
 
 function down-test-cluster() {
+    if [[ -n "${CIRCLE_JOB:-}" ]]; then
```
Comment (jaypipes) [inline scripts/lib/cluster.sh]: Why would you want to disable the cluster delete when running in CI?

File: scripts/lib/cluster.sh [hunk 2]
```diff
@@ -28,9 +32,14 @@ function up-test-cluster() {
...
+        $TESTER_PATH eks create config --path $CLUSTER_CONFIG 1>&2
+
+    if [[ -n "${CIRCLE_JOB:-}" ]]; then
+        $TESTER_PATH eks create cluster --path $CLUSTER_CONFIG || (echo "failed!" && exit 1)
+    else
+        echo -n "Creating cluster $CLUSTER_NAME (this may take ~20 mins. details: tail -f $CLUSTER_MANAGE_LOG_PATH)... "
+        $TESTER_PATH eks create cluster --path $CLUSTER_CONFIG 2>> $CLUSTER_MANAGE_LOG_PATH 1>&2 ||
+            (echo "failed. Check $CLUSTER_MANAGE_LOG_PATH." && exit 1)
+        echo "ok."
+    fi
```
Comment (jaypipes) [inline scripts/lib/cluster.sh]: I'm confused why you're changing the above depending on whether it's running in CI or not?

File: scripts/run-integration-tests.sh [hunk 3]
```diff
@@ -138,10 +138,25 @@ cp "$BASE_CONFIG_PATH" "$TEST_CONFIG_PATH"
...
+GO111MODULE=on go test -v -timeout 0 ./... --kubeconfig=$KUBECONFIG --ginkgo.focus="\[cni-integration\]" --ginkgo.skip="\[Disruptive\]" \
+    --assets=./assets
+TEST_PASS=$?
+popd
+
+echo "*******************************************************************************"
+echo "Updating CNI to image $IMAGE_NAME:$TEST_IMAGE_VERSION"
+$KUBECTL_PATH apply -f "$TEST_CONFIG_PATH"
+
+echo "Sleping for 110s"
```
Comment (jaypipes) [inline scripts/run-integration-tests.sh]: "Sleeping"

File: scripts/run-integration-tests.sh [hunk 4]
```diff
@@ -138,10 +138,25 @@ cp "$BASE_CONFIG_PATH" "$TEST_CONFIG_PATH"
...
+echo "TODO: Poll and wait for updates to complete instead!"
+sleep 110
```
Comment (jaypipes) [inline scripts/run-integration-tests.sh]: why 110 seconds? is that a magic value? :)

File: scripts/lib/cluster.sh (hunk 1 above)
Comment (mogren) [reply scripts/lib/cluster.sh]: Just making it more verbose. See comment below.

File: scripts/lib/cluster.sh (hunk 2 above)
Comment (mogren) [reply scripts/lib/cluster.sh]: It's just changing the logging output. By default, CircleCI will terminate any test after 10 minutes fi there is no console output. That can be changed by setting `no_output_timeout` for the job, but when you have to wait 40 minutes it's more interesting to see what is going on.

File: scripts/run-integration-tests.sh (hunk 4 above)
Comment (mogren) [reply scripts/run-integration-tests.sh]: Yes, for now. The test cluster has 3 nodes, and the rolling update is using the default grace period of 30s, so it takes around 30s to shut down each node, and another 5-10 seconds to start up. It's ok to start the test once all old ipamd pods have shut down.

File: scripts/lib/cluster.sh (hunk 2 above)
Comment (jaypipes) [reply scripts/lib/cluster.sh]: ack, makes sense. ++

**Pull request merged**